labs/a1.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Advanced Lab 1
+layout: lab
+---
+
+Lecture: Thursday 2018-09-13
+Released: Monday 2018-09-17
+Due: Monday 2018-10-01
+
+# Advanced section, Lab 1
+
+### Grading note
+Labs are graded on completion. Treat this lab as seeds of exploration instead 
+of just a grade. If you don't pass on the first submission, you can have it 
+checked off in-person by a decal facilitator. As this is the first lab, it
+is intentionally simple and easy to complete, but should introduce you to some
+fun bash features you may not have encountered before, such as loops and shell
+expansions.
+
+
+### Workflow
+This lab can be done on your own UNIX-like machine, or you can ssh into 
+`tsunami.ocf.berkeley.edu` using your OCF account to finish the lab there. As 
+always, `man` and Google will be your friends.
+
+## Question 1
+
+Using Bash functions, write a script that generates a user-specified number of files of
+user-specified size filled with random content.
+
+e.g.
+
+```bash
+$ ./mkrandom 10 100  # create 10 100 byte random files
+$ ls -lAh
+total 44K
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 1
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 10
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 2
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 3
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 4
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 5
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 6
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 7
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 8
+-rw-r--r-- 1 abizer ocf  100 Sep 16 21:57 9
+-rwxr-xr-x 1 abizer ocf  147 Sep 16 21:56 mkrandom
+```
+
+You may want to look into `dd` and the `iflag=fullblock` argument,
+`seq`, and `/dev/random`.
```

Comment:
I left these like this because the basic google searches for these terms should cover their usage to the extent we expect advanced students to be able to understand

Reply:
Looking at this again and you could be right, I'll leave it up to you to decide.